### ISSUE TITLE
How to run inferience for a single image (not included in the training set)

### ISSUE THREAD

**Issue by sjcotto** [miccunifi/ladi-vton#10]
Hey, great project

How to run inferience for a single image (not included in the training set)

Team have plans for a google colab?

**Comment by ykk648** [comment]
working on it  https://github.com/sangyun884/HR-VITON/issues/45

**Comment by ABaldrati** [comment]
Hi @sjcotto 
Thank you for reaching out and showing interest in our work! 
Regarding your request for a Google Colab demonstrating the specific functionality you mentioned, we currently do not have plans to release such a Colab.

To run inference on an image, you will need the following components:
- Model image: This is the image you want to use as a reference for the desired output.
- In-shop garment image: This is the image of the clothing item you want to transfer onto the model image.- 
- Model image pose: Our system utilizes 18 key points to represent the pose of the model in the image. 
- Model image label map: This is a label map specific to the model image, indicating different regions of the body.

To perform inference, you would need to process the model image using the same code as our custom PyTorch dataset. Once you have processed the model image with these ingredients, you should be able to run the inference successfully.

For a guide on starting this process, @ykk648 has provided a useful resource that you can refer to.

Alberto












We are not planning to release any Google colab showing what you are asking for.
For running inference on an image you would need:
- The model image
- The in-shop garment image
- The model image pose (we use 18 key points)
- The model image label map

With these ingredients, you should then process the model image using the same code of our custom Pytorch dataset and then you should be able to run inference.

@ykk648 provide a nice guide to start such a process!

**Comment by kadirnar** [comment]
> 

I want to test this model. I want to do it by giving 1 image of people and clothes. But I don't want to test it on the dataset. Which libraries should I use to pose and extract the feature map? Can I do it using this library?

**Comment by kadirnar** [comment]
│ │ ├── masks
│ │ │ ├── [013563_1.png| 013564_1.png | ...]
│ │ ├── keypoints
│ │ │ ├── [013563_2.json | 013564_2.json | ...]
│ │ ├── label_maps
│ │ │ ├── [013563_4.png | 013564_4.png | ...]
│ │ ├── skeletons
│ │ │ ├── [013563_5.jpg | 013564_5.jpg | ...]
│ │ ├── dense
│ │ │ ├── [013563_5.png | 013563_5_uv.npz | 013564_5.png | 013564_5_uv.npz | ...]


How can I make these using custom images?
